src/main/java/eu/europeana/api/recommend/web/RecommendController.java

```diff
@@ -143,8 +142,9 @@ public ResponseEntity recommendEntity(
             @RequestHeader(value = HttpHeaders.AUTHORIZATION, required = false) String authToken)
             throws RecommendException {
         String apikey = checkCredentials(authToken, wskey);
-
-        Mono result = recommendService.getRecommendationsForEntity(type, id, pageSize, authToken, apikey);
+        String entityId = recommendService.buildEntityId(type, id);
+        String requestBody = recommendService.getEntityRecommendationRequest(entityId, apikey);
+        Mono result = recommendService.getRecommendationsForEntity(entityId, requestBody, pageSize, authToken, apikey);
```

Comment:
The job of a controller is to validate input and send back output. All other code should be part of a service as much as possible. It's the service's job to do the dirty work. So to improve structure I would propose to alter the `getRecommendationsForEntity` method in the service and make sure that this method:
1. generates the proper entityId
2. creates a requestBody to sent to entity api
3. sends the request to entity api and returns the proper response to the controller

Reply:
done